site/source/docs/compiling/Dynamic-Linking.rst

```diff
@@ -77,19 +77,17 @@ the problem as follows: There are two types of shared modules:
 
 A project should contain **exactly one** main module. It can then be
 linked at runtime to multiple side modules. This model also makes other
-things simpler, like only the singleton main module has the general
-JavaScript environment setup code to connect to the web page and so
-forth; side modules contain just the pure compiled WebAssembly and
-nothing more.
+things simpler, like only the singleton main module includes the
+JavaScript environment and side modules are pure WebAssembly modules.
 
-The one tricky aspect to this design is that a side module might need a
-system library that the main doesn’t know about. See the section on
+The one tricky aspect to this design is that a side module might depend on a
+system library that the main module did not depend on. See the section on
```

Comment:
how about "function in a system library such as pow from libc" (or is it actually a whole library that's being referred to?)

Reply:
It actually works on per library basis, since MAIN_MODULE=1 links all libraries with `--whole-archive`.